crypto/ml_dsa/ml_dsa.c

```diff
@@ -271,25 +327,33 @@ int ml_dsa_extmu_65_verify_internal(const uint8_t *public_key /* IN */,
 
 int ml_dsa_87_keypair(uint8_t *public_key   /* OUT */,
                       uint8_t *private_key  /* OUT */) {
+  boringssl_ensure_ml_dsa_self_test();
   ml_dsa_params params;
   ml_dsa_87_params_init(&params);
   return (ml_dsa_keypair(&params, public_key, private_key) == 0);
 }
 
-int ml_dsa_87_pack_pk_from_sk(uint8_t *public_key          /* OUT */,
-                              const uint8_t *private_key   /* IN  */) {
+int ml_dsa_87_keypair_internal(uint8_t *public_key   /* OUT */,
+                               uint8_t *private_key  /* OUT */,
+                               const uint8_t *seed   /* IN */) {
+  boringssl_ensure_ml_dsa_self_test();
+  return ml_dsa_87_keypair_internal_no_self_test(public_key, private_key, seed);
+}
 
+int ml_dsa_87_keypair_internal_no_self_test(uint8_t *public_key   /* OUT */,
+                                            uint8_t *private_key  /* OUT */,
+                                            const uint8_t *seed   /* IN */) {
   ml_dsa_params params;
   ml_dsa_87_params_init(&params);
-  return ml_dsa_pack_pk_from_sk(&params, public_key, private_key) == 0;
+  return ml_dsa_keypair_internal(&params, public_key, private_key, seed) == 0;
 }
```

Comment:
Do we need to do this? We only need the _no_self_test functions to avoid a infinite loop of the self test calling itself. Here your self test only tests ML-DSA44 so we should only need it for that size. The other 65/87 functions can call `boringssl_ensure_ml_dsa_self_test()` and it will run the 44 no self test version without hitting a loop.

Reply:
You are right! They aren't needed, I got rid of them. [6a296df](https://github.com/aws/aws-lc/pull/2148/commits/6a296df58e950ee71d01cea75f8815723ac9abe5)